sw/device/lib/runtime/otbn.c

```diff
@@ -4,16 +4,96 @@
 
 #include "sw/device/lib/runtime/otbn.h"
 
+#include "sw/device/lib/base/freestanding/limits.h"
 #include "sw/device/lib/base/mmio.h"
 #include "sw/device/lib/dif/dif_base.h"
 #include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
 #include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
 
 /**
  * Data width of big number subset, in bytes.
  */
 const int kOtbnWlenBytes = 256 / 8;
 
+/**
+ * Otbn context to be used by the isr handler, it should be assigned with the
+ * current otbn context pointer before enabling the irq.
+ */
+static otbn_t *otbn_ctx;
```

Comment:
Maybe give this a name that reflects where it's used (e.g. `otbn_ctx_for_irq`)? Just to avoid confusion with this global variable and the local ones that are passed around within the other functions.